e2e/test/scenarios/custom-column/cc-shortcuts-combine.cy.spec.js

```diff
@@ -111,6 +111,26 @@ describe("scenarios > question > custom column > expression shortcuts > combine"
       cy.findByLabelText("Separator").should("have.value", "");
     });
   });
+
+  it("should be possible to edit a previous stages' columns when there is an aggregation (metabase#43226)", () => {
```

Comment:
I found more combine/extract-related issues. Not sure if they are or aren't caused by this PR but I found them all in this branch.

1. Create a new question based on Orders table
2. Add Count aggregation
3. Add breakout by Discount (auto bin)
4. Add breakout by Product > Category
5. :x:  Try to add a custom column after aggregations - "Extract column" shortcut is displayed even though there are no sub-options for it
6. Visualize (without actually adding the column in step 5)
7. Notice that "+" button is not displayed (correct)
8. Switch to raw data view
9. :x: "+" button is displayed, nothing happens after clicking it
10. Switch back to visualization view
11. :x:  "+" button is now displayed, after clicking it there are options based on columns pre-aggregation step (I'm pretty sure this is caused by this PR)
12. Switch back to raw data view
13. Click "Product -> Category" column header
14. Choose "Combine columns" and apply the defaults
15. Switch back to visualization view
16. Click "+" button
17. :x: "Extract part of column" option is shown, even though there are no sub-options
18. Click "< Select column to extract from" to go back to the previous menu
19. :x: Menu closes instead of going to the previous level
    - I guess it's ok if menu closes, but we should not show "<" icon indicating that it'll take you back to the previous menu

https://github.com/metabase/metabase/assets/6830683/e4e4f276-0c83-4de6-bc63-506fe11caeac